### ISSUE TITLE
DockMenu menu (context menu)

### ISSUE THREAD

**Issue by Moosems** [tukaan/tukaan#84]
Important: when using MacOS having a good DockMenu menu is incredibly important. I propose that we create a good system for this as it does not seem to be built into tk. See my [stackoverflow question](https://stackoverflow.com/questions/72834439/open-new-window-tkinter-tk-taskbar-dock/72836549#72836549) for more detail. This is also an issue for windows so setting up a good system is important to get done at some point.

**Comment by rdbende** [comment]
Yes, that would be a good addition for Mac and Windows too (I'd love to see it on Linux, but there are so many different desktop environments and task managers that it would be impossible to make it compatible with all of them).
The other thing is the progressbar on the dock icons (I don't know if there is one on macOS). I think both I think on Windows both can be done using `ctypes` winapi. I don't know much about mac stuff, but we have [tukaan/Idared](https://github.com/tukaan/Idared), IMO it could be integrated into that.

**Comment by Moosems** [comment]
I have never seen a loading bar on the Mac dock. The winapi seems to made directly for windows but if done right I think Mac can be handled through pyobjc.

**Comment by Moosems** [comment]
@rdbende check the stackoverflow question as there is a good pyobjc reference given

**Comment by Moosems** [comment]
Any updates?

**Comment by Moosems** [comment]
Any way I can help?

**Comment by sumeshir26** [comment]
What is Idared? I cant seem to access it, is it private?

**Comment by Moosems** [comment]
@sumeshir26 Good question. I can't access it so it's probably @rdbende's private repo

**Comment by rdbende** [comment]
Lol, I though it's public.

**Comment by rdbende** [comment]
https://github.com/tukaan/Idared
